fix: cleanup lint warnings

ludtkemorgan opened the pull request:
This doesn't address any specific issue, but I noticed we had a decent amount of unused variables/imports. This PR cleans up the current warnings and also tightens the max number of warnings to our current level.

I did not fix any of the "any" types or the missing dependencies in hooks. These will require more testing/knowledge of that area of code

## How Can This Be Tested/Reviewed?

All tests should still work as well building and starting the app

## Author Checklist:

- [ ] Added QA notes to the issue with applicable URLs
- [ ] Reviewed in a desktop view
- [ ] Reviewed in a mobile view
- [ ] Reviewed considering accessibility
- [ ] Added tests covering the changes
- [ ] Made corresponding changes to the documentation
- [ ] Ran `yarn generate:client` and/or created a migration when required

## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates

emilyjablonski reviewed (approved):
LGTM! Curious why there were lock file changes when it doesn't look like any packages changed?

ludtkemorgan commented:
> LGTM! Curious why there were lock file changes when it doesn't look like any packages changed?

@emilyjablonski I tried upgrading eslint to see if it would solve the weird issue where two imports they think they aren't being used even though they are. But the major upgrade required a full rewrite of our eslint config file so I reverted it. However since we use the `^` for most dependencies when I reverted it yarn pulled in newer minor versions
